Tests/NIOHPACKTests/IntegerCodingTests.swift

```diff
@@ -0,0 +1,158 @@
+        //  set the top bit of all but the last part to get our remaining output bytes:
+        //      10000000 10000000 10000000 10000000 10000000 10000000 10000000 10000000 10000000 00000001
+        XCTAssertEqual(data.count, 11)
+        XCTAssertEqual(data, [1, 128, 128, 128, 128, 128, 128, 128, 128, 128, 1])
+        
+        // something similar, which uses an 8-bit prefix and still produces lots of zero bits:
+        // for those interested: this is the previous value + 254; thus only the first byte should differ
+        data = encodeIntegerToArray(9223372036854776063, prefix: 8)
+        
+        // calculations:
+        //  subtract prefix:
+        //      9223372036854776063 - 255 = 9223372036854775808 or 1000 (0000 x15)
+        //  seven bits at a time, grouping from least significant bit:
+        //      (000000)1 0000000 0000000 0000000 0000000 0000000 0000000 0000000 0000000 0000000
+        //  swap these around:
+        //      0000000 0000000 0000000 0000000 0000000 0000000 0000000 0000000 0000000 0000001
+        //  set the top bit of all but the last part to get our remaining output bytes:
+        //      10000000 10000000 10000000 10000000 10000000 10000000 10000000 10000000 10000000 00000001
+        XCTAssertEqual(data.count, 11)
+        XCTAssertEqual(data, [255, 128, 128, 128, 128, 128, 128, 128, 128, 128, 1])
+    }
+    
+    func testIntegerDecoding() throws {
+        // any bits above the prefix amount shouldn't affect the outcome.
+        XCTAssertEqual(try decodeInteger(from: [0b00001010], prefix: 5), 10)
+        XCTAssertEqual(try decodeInteger(from: [0b11101010], prefix: 5), 10)
+        
+        XCTAssertEqual(try decodeInteger(from: [0b00011111, 154, 10], prefix: 5), 1337)
+        XCTAssertEqual(try decodeInteger(from: [0b11111111, 154, 10], prefix: 5), 1337)
+        
+        XCTAssertEqual(try decodeInteger(from: [0b00101010], prefix: 8), 42)
+        
+        // Now some larger numbers:
+        XCTAssertEqual(try decodeInteger(from: [255, 129, 254, 255, 255, 255, 255, 255, 255, 33], prefix: 8), 2449958197289549824)
+        XCTAssertEqual(try decodeInteger(from: [1, 255, 255, 255, 255, 255, 255, 255, 255, 33], prefix: 1), 2449958197289549824)
+        XCTAssertEqual(try decodeInteger(from: [1, 254, 255, 255, 255, 255, 255, 255, 255, 255, 1], prefix: 1), UInt(UInt64.max))
+        
+        // lots of zeroes: each 128 yields zero
+        XCTAssertEqual(try decodeInteger(from: [1, 128, 128, 128, 128, 128, 128, 128, 128, 128, 1], prefix: 1), 9223372036854775809)
+        
+        // almost the same bytes, but a different prefix:
+        XCTAssertEqual(try decodeInteger(from: [255, 128, 128, 128, 128, 128, 128, 128, 128, 128, 1], prefix: 8), 9223372036854776063)
+        
+        // now a silly version which should never have been encoded in so many bytes
+        XCTAssertEqual(try decodeInteger(from: [255, 129, 128, 128, 128, 128, 128, 128, 128, 0], prefix: 8), 256)
+    }
+}
```

Comment:
This testing is good, but I'd love to see even more. Are you open to bringing in [these test fixtures](https://github.com/http2jp/hpack-test-case) and using them to generate tests? We probably want to test that a) we can decode any of the encoded samples and get the expected result, and b) that we can encode the samples, decode them, and get back to the expected result.

That would give us loads of test coverage, which would be awesome.

Reply:
Rawr. Definitely.

Reply:
What's your preference for this? Have the regular `XCTest` implementation use these, or have a separate verification script that runs through them, leaving the XCTests for something simpler?

Reply:
Yeah, this is a real future growth area for Swift tooling: the inability to dynamically turn each of those into an XCTest test case makes me extremely sad.

That said, I think I'd still prefer that they lived in XCTests. We probably get the best outcome by using custom error messages on each of the `XCTAssert` methods that encode the identity of the test case that failed the specific assertion, so that we can more easily track down failures.